packages/@aws-cdk/aws-route53-targets/test/bucket-website-target.test.ts

```diff
@@ -93,3 +94,46 @@ test('throws if bucket website hosting is unavailable (cn-northwest-1)', () => {
     });
   }).toThrow(/Bucket website target is not supported/);
 });
+
+
+test("throws if bucket name doesn't match full record domain name: new Bucket", () => {
+  // GIVEN
+  const app = new App();
+  const stack = new Stack(app, 'test', {env: {region: 'us-east-1'}});
+
+  // FIXME bucketName is unresolved?
+  const bucketWebsite = new s3.Bucket(stack, 'Bucket', { bucketName: ['different', bucketName].join('.') });
```

Comment:
Any idea on how I could access `bucketName` there? The property is a token once I access it in the `validate` method